core/src/main/java/com/linecorp/armeria/common/SerializationFormat.java

```diff
@@ -142,9 +247,109 @@ public String uriText() {
     }
 
     /**
-     * Returns the {@link MediaType} of this format.
+     * Returns the primary {@link MediaType} of this format.
      */
     public MediaType mediaType() {
-        return mediaType;
+        return primaryMediaType;
+    }
+
+    /**
+     * Returns the media types accepted by this format.
+     */
+    public Set<MediaType> mediaTypes() {
+        return allMediaTypes;
+    }
+
+    /**
+     * Returns whether the specified {@link MediaType} is accepted by any of the {@link #mediaTypes()}
+     * defined by this format.
+     */
+    public boolean isAccepted(MediaType mediaType) {
+        requireNonNull(mediaType, "mediaType");
+
+        // Similar to what MediaType.is(MediaType) does except that
+        // this one compares the parameters case-insensitively.
+        for (MediaType type : allMediaTypes) {
+            if (!type.type().equals(mediaType.type()) ||
+                !type.subtype().equals(mediaType.subtype())) {
+                continue;
+            }
+
+            final ImmutableListMultimap<String, String> requiredParameters = type.parameters();
+            final ImmutableListMultimap<String, String> actualParameters = mediaType.parameters();
+            if (containsAllParameters(requiredParameters, actualParameters)) {
+                return true;
+            }
+        }
+
+        return false;
+    }
+
+    private static boolean containsAllParameters(Multimap<String, String> requiredParameters,
+                                                 Multimap<String, String> actualParameters) {
+
+        if (requiredParameters.isEmpty()) {
+            return true;
+        }
+
+        for (Entry<String, Collection<String>> requiredEntry : requiredParameters.asMap().entrySet()) {
+            final Collection<String> requiredValues = requiredEntry.getValue();
+            final Collection<String> actualValues = actualParameters.get(requiredEntry.getKey());
+
+            assert !requiredValues.isEmpty();
+            if (actualValues.isEmpty()) {
+                // Does not contain any required values.
+                return false;
+            }
+
+            if (!containsAllRequiredValues(requiredValues, actualValues)) {
+                return false;
+            }
+        }
+
+        return true;
+    }
+
+    private static boolean containsAllRequiredValues(Iterable<String> requiredValues,
+                                                     Iterable<String> actualValues) {
+        boolean containsAllRequiredValues = true;
+        for (String requiredValue : requiredValues) {
```

Comment:
Fixed. Used simple for loop given that this method can be called pretty often, although stream makes code much simpler..